easypermissions/src/main/java/pub/devrel/easypermissions/helper/PermissionHelper.java

```diff
@@ -14,39 +13,38 @@
 /**
  * Delegate class to make permission calls based on the 'host' (Fragment, Activity, etc).
  */
-@RestrictTo(RestrictTo.Scope.LIBRARY)
 public abstract class PermissionHelper<T> {
 
-    private static final String TAG = "PermissionHelper";
-
     private T mHost;
 
     @NonNull
-    public static PermissionHelper newInstance(Activity host) {
+    public static PermissionHelper<Activity> newInstance(Activity host) {
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-            return new LowApiPermissionsHelper(host);
+            return new LowApiPermissionsHelper<>(host);
         }
 
-        if (host instanceof AppCompatActivity) {
-            return new AppCompatActivityPermissionHelper((AppCompatActivity) host);
-        } else {
+        if (host instanceof AppCompatActivity)
+            //noinspection unchecked Hmmm... not sure what's going on here
+            return (PermissionHelper<Activity>) (Object)
+                    new AppCompatActivityPermissionHelper((AppCompatActivity) host);
+        else {
             return new ActivityPermissionHelper(host);
         }
     }
 
     @NonNull
-    public static PermissionHelper newInstance(Fragment host) {
+    public static PermissionHelper<Fragment> newInstance(Fragment host) {
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-            return new LowApiPermissionsHelper(host);
+            return new LowApiPermissionsHelper<>(host);
```

Comment:
Surprised to see the `<>` here, do we get a warning without that?

Reply:
Yeah, I added a type `T` to the helper.